internal/config/config.go

```diff
@@ -17,66 +17,52 @@ package config
-		Name:        "config/vars_total",
-		Measure:     cfgVarCount,
-		Description: "The number of config vars read during initialization",
-		Aggregation: view.Count(),
+// Read sets default to a viper instance and read user config to override these defaults.
+func Read() (View, error) {
+	var err error
+	// read configs from config/default/matchmaker_config_default.yaml
+	// matchmaker_config_default provides default values for all of the possible tunnable parameters in Open Match
+	dcfg := viper.New()
+	dcfg.SetConfigType("yaml")
+	dcfg.AddConfigPath(".")
+	dcfg.AddConfigPath("config/default")
+	dcfg.SetConfigName("matchmaker_config_default")
+	err = dcfg.ReadInConfig()
+	if err != nil {
+		log.Fatalf("fatal error reading default config file, desc: %s", err.Error())
 	}
-)
 
-// Read reads a config file into a viper.Viper instance and associates environment vars defined in
-// config.envMappings
-func Read() (View, error) {
+	// read configs from config/override/matchmaker_config_default.yaml
+	// matchmaker_config_override overrides default values specified in matchmaker_config_default
 	cfg := viper.New()
-	// Viper config management initialization
-	// Support either json or yaml file types (json for backwards compatibility
-	// with previous versions)
-	cfg.SetConfigType("json")
-	cfg.SetConfigType("yaml")
-	cfg.AddConfigPath(".")
-	cfg.AddConfigPath("config/global")
-	cfg.SetConfigName("global_config")
-	err := cfg.ReadInConfig()
-	if err != nil {
-		logger.WithError(err).Fatal("Fatal error reading config file")
+
+	// set defaults for cfg using settings in dcfg
+	for k, v := range dcfg.AllSettings() {
+		cfg.SetDefault(k, v)
 	}
 
-	cfg.AddConfigPath("config/om")
-	cfg.SetConfigName("matchmaker_config")
-	// Read in config file using Viper
-	err = cfg.MergeInConfig()
+	cfg.SetConfigType("yaml")
+	cfg.AddConfigPath(".")
+	cfg.AddConfigPath("config/override")
+	cfg.SetConfigName("matchmaker_config_override")
+	err = cfg.ReadInConfig()
 	if err != nil {
-		logger.WithError(err).Fatal("Fatal error reading config file")
+		log.Fatalf("fatal error reading override config file, desc: %s", err.Error())
 	}
 
 	// Look for updates to the config; in Kubernetes, this is implemented using
-	// a ConfigMap that is written to the matchmaker_config.yaml file, which is
+	// a ConfigMap that is written to the matchmaker_config_override.yaml file, which is
 	// what the Open Match components using Viper monitor for changes.
 	// More details about Open Match's use of Kubernetes ConfigMaps at:
 	// https://open-match.dev/open-match/issues/42
 	cfg.WatchConfig() // Watch and re-read config file.
 	// Write a log when the configuration changes.
 	cfg.OnConfigChange(func(event fsnotify.Event) {
-		logger.WithFields(logrus.Fields{
-			"filename":  event.Name,
-			"operation": event.Op,
-		}).Info("Server configuration changed.")
+		log.Printf("Server configuration changed, operation: %v, filename: %s", event.Op, event.Name)
 	})
 	return cfg, err
```

Comment:
Returning a config or error should be mutually exclusive.  return cfg, nil, if err is known to be nil.